packages/formation-react/src/components/SortableTable/SortableTable.jsx

```diff
@@ -1,19 +1,62 @@
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
 import classNames from 'classnames';
 
-class SortableTable extends React.Component {
-  constructor(props) {
-    super(props);
-    this.makeHeader = this.makeHeader.bind(this);
-    this.makeRow = this.makeRow.bind(this);
-  }
+class SortableTable extends Component {
+  static propTypes = {
+    // Used to pass custom classes.
+    className: PropTypes.string,
 
-  handleSort(value, order) {
-    return () => this.props.onSort(value, order);
+    /**
+     * Field value to sort by in either ascending or descending order.
+     * The `value` must be one of the values in the `fields` prop.
+     */
+    currentSort: PropTypes.shape({
+      value: PropTypes.string.isRequired,
+      order: PropTypes.oneOf(['ASC', 'DESC']),
+    }).isRequired,
+
+    // Mappings of header labels to properties on the objects in `data`.
+    fields: PropTypes.arrayOf(
+      PropTypes.shape({
+        label: PropTypes.string.isRequired,
+        value: PropTypes.string.isRequired,
+      }),
+    ).isRequired,
+
+    /**
+     * Each object represents data for a row.
+     * An optional class may be provided to style specific rows.
+     */
+    data: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+        rowClass: PropTypes.string,
+        values: PropTypes.arrayOf(
+          PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+        ),
+      }),
+    ).isRequired,
+
+    // A callback for when a header is clicked.
+    onHeaderClick: PropTypes.func,
+  };
+
+  onHeaderClick = (value, order) => () => {
+    const { onHeaderClick } = this.props;
+
+    // This is a legacy prop that is being renamed to `onHeaderClick`.
+    if (onSort) {
+      console.error('WARNING: The `onSort` prop is now deprecated in favor of the `onHeaderClick` prop for <SortableTable />.');
+    }
```

Comment:
Since we're bumping the major version, we can remove support for `onSort`.

Reply:
All this does is in case an `onSort` prop is provided, they will get an error noting that it's now deprecated so it doesn't just silently fail on them :)

Reply:
I think it's okay for it to fail silently since we're bumping the major version. React will warn them about an unrecognized prop anyway.

Reply:
I don't think PropTypes catches props that aren't included (unrecognized props), but will change this so it fails silently! 👍